### ISSUE TITLE
Flutter 3.4.0+ regression in code coverage computation

### ISSUE THREAD

**Issue by tsinis** [flutter/flutter#115719]
## Steps to Reproduce

Every version of Flutter with Dart SDK v2.19 is out of sync with this cherry-picked fix I've reported in [dart-lang/sdk](https://github.com/dart-lang/sdk/issues/49887). It was fixed in Dart v2.18.3 and working correctly in the current stable release of Flutter (Dart v2.18.4). But not on Flutter's master and beta (since v 3.4.0+).

1. `flutter channel beta` 
2. `flutter upgrade` 
3. `git clone https://github.com/tsinis/mixins_tests_bug.git` 
4. `cd mixins_tests_bug` 
5. `git checkout test/dart-2.19-in-flutter` 
6. `flutter test --coverage` 

As a bonus run, but this is not necessary: `genhtml coverage/lcov.info -o coverage/html/`

**Expected results:** 

Proper coverage in mixins in pure Dart 2.19:
https://github.com/tsinis/mixins_tests_bug/blob/test/dart-2-19-without-flutter/coverage/lcov.info

**Actual results:** 

Wrong coverage in mixins in Dart 2.19 that is coming with Flutter SDK.
https://github.com/tsinis/mixins_tests_bug/blob/test/dart-2.19-in-flutter/coverage/lcov.info

<details>
<summary>Code sample</summary>

```dart
mixin CoverageBug {
  bool boolean = false;

  bool wrongCoverage() {
    print('starting method');
    const i = 1;
    final x = 2;
    var y = 3;
    boolean = !(i == x + y);
    print('triggered all those lines in test for coverage');

    return boolean;
  }
}
```

```dart
import 'package:flutter_test/flutter_test.dart';
import 'package:mixins_tests_bug/main.dart';

class _MixinTest with CoverageBug {}

void main() => group('$CoverageBug', () {
      final mixinBugTest = _MixinTest();

      test('wrongCoverage()', () {
        expect(mixinBugTest.boolean, isFalse);
        expect(mixinBugTest.wrongCoverage(), isTrue);
        expect(mixinBugTest.boolean, isTrue);
      });
    });
```

</details>

<details>
  <summary>Logs</summary>

```
[   +6 ms] executing: sysctl hw.optional.arm64
[   +4 ms] Exit code 0 from: sysctl hw.optional.arm64
[        ] hw.optional.arm64: 1
[        ] executing: [/Users/roman.cinis@caronsale.de/fvm/versions/beta/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +10 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 75927305ff855f76a9ef704f9b4a86fa2fce7292
[        ] executing: [/Users/roman.cinis@caronsale.de/fvm/versions/beta/] git tag --points-at 75927305ff855f76a9ef704f9b4a86fa2fce7292
[  +15 ms] Exit code 0 from: git tag --points-at 75927305ff855f76a9ef704f9b4a86fa2fce7292
[        ] 3.6.0-0.1.pre
[   +5 ms] executing: [/Users/roman.cinis@caronsale.de/fvm/versions/beta/] git rev-parse --abbrev-ref --symbolic @{upstream}
[   +8 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] origin/beta
[        ] executing: [/Users/roman.cinis@caronsale.de/fvm/versions/beta/] git ls-remote --get-url origin
[   +7 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +14 ms] executing: [/Users/roman.cinis@caronsale.de/fvm/versions/beta/] git rev-parse --abbrev-ref HEAD
[   +9 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] beta
[        ] executing: sw_vers -productName
[   +8 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[   +9 ms] Exit code 0 from: sw_vers -productVersion
[        ] 13.0.1
[        ] executing: sw_vers -buildVersion
[   +9 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 22A400
[        ] executing: uname -m
[   +3 ms] Exit code 0 from: uname -m
[        ] arm64
[   +5 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +33 ms] executing: /Users/roman.cinis@caronsale.de/Library/Android/sdk/platform-tools/adb devices -l
[   +5 ms] executing: sysctl hw.optional.arm64
[   +3 ms] Exit code 0 from: sysctl hw.optional.arm64
[        ] hw.optional.arm64: 1
[        ] executing: /usr/bin/arch -arm64e xcrun xcodebuild -version
[ +114 ms] Exit code 0 from: /usr/bin/arch -arm64e xcrun xcodebuild -version
[        ] Xcode 14.1
           Build version 14B47b
[   +1 ms] executing: /usr/bin/arch -arm64e xcrun xcdevice list --timeout 2
[        ] /usr/bin/arch -arm64e xcrun simctl list devices booted iOS --json
[        ] executing: /usr/bin/arch -arm64e xcrun simctl list devices booted iOS --json
[   +1 ms] executing: /usr/bin/arch -arm64e xcrun simctl list devices booted
[ +100 ms] Exit code 0 from: /usr/bin/arch -arm64e xcrun simctl list devices booted
[        ] == Devices ==
           -- iOS 13.7 --
           -- iOS 16.1 --
[  +14 ms] List of devices attached
           emulator-5554          device product:sdk_gphone64_arm64 model:sdk_gphone64_arm64 device:emu64a transport_id:1
[  +90 ms] {
             "devices" : {
               "com.apple.CoreSimulator.SimRuntime.iOS-16-1" : [

               ],
               "com.apple.CoreSimulator.SimRuntime.iOS-13-7" : [

               ]
             }
           }
[+2335 ms] [
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad14,3",
               "identifier" : "B2F1F461-E66A-4AB8-A4B7-2C9053FFD623",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.ipad-pro-11-4th-1",
               "modelName" : "iPad Pro (11-inch) (4th generation)",
               "name" : "iPad Pro (11-inch) (4th generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad14,5",
               "identifier" : "358572AF-2D67-4EFF-BCEF-18BF6F6F94D1",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.ipad-pro-12point9-6th-1",
               "modelName" : "iPad Pro (12.9-inch) (6th generation)",
               "name" : "iPad Pro (12.9-inch) (6th generation)"
             },
             {
               "modelCode" : "iPhone8,4",
               "simulator" : false,
               "modelName" : "iPhone SE (Model A1662, A1723, A1724)",
               "error" : {
                 "code" : -13,
                 "failureReason" : "",
                 "underlyingErrors" : [
                   {
                     "code" : 4,
                     "failureReason" : "",
                     "description" : "iPhone (Roman) is locked.",
                     "recoverySuggestion" : "To use iPhone (Roman) with Xcode, unlock it.",
                     "domain" : "DVTDeviceIneligibilityErrorDomain"
                   }
                 ],
                 "description" : "iPhone (Roman) is not connected",
                 "recoverySuggestion" : "Xcode will continue when iPhone (Roman) is connected and unlocked.",
                 "domain" : "com.apple.platform.iphoneos"
               },
               "operatingSystemVersion" : "15.6 (19G71)",
               "identifier" : "a141e30024dbc94fc22dcbee4b9bd2418aaef54e",
               "platform" : "com.apple.platform.iphoneos",
               "architecture" : "arm64",
               "interface" : "usb",
               "available" : false,
               "name" : "iPhone (Roman)",
               "modelUTI" : "com.apple.iphone-se-a1662-dcdede"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone12,5",
               "identifier" : "74A79DEC-408D-44D0-9C4F-4AF028A90135",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.iphone-11-pro-max-1",
               "modelName" : "iPhone 11 Pro Max",
               "name" : "iPhone 11 Pro Max"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone14,8",
               "identifier" : "16ED7D91-4912-4096-B873-5F9AF961CD1D",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.iphone-14-plus-1",
               "modelName" : "iPhone 14 Plus",
               "name" : "iPhone 14 Plus"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad11,3",
               "identifier" : "54FE019B-C3CD-4085-B4FC-F51475399077",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.ipad-air3-wifi-1",
               "modelName" : "iPad Air (3rd generation)",
               "name" : "iPad Air (3rd generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad8,12",
               "identifier" : "5C96B117-3C74-41BA-A5F7-563314A457CC",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.ipad-pro-12point9-4th-1",
               "modelName" : "iPad Pro (12.9-inch) (4th generation)",
               "name" : "iPad Pro (12.9-inch) (4th generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone15,2",
               "identifier" : "BE2C59EF-7B7A-459A-827F-2FDCACE4FA9B",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.iphone-14-pro-1",
               "modelName" : "iPhone 14 Pro",
               "name" : "iPhone 14 Pro"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad14,1",
               "identifier" : "9606569C-946D-4197-B322-CE2F01348506",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.ipad-mini6-1",
               "modelName" : "iPad mini (6th generation)",
               "name" : "iPad mini (6th generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone12,8",
               "identifier" : "CCFCCDEC-8A3C-471C-8FA2-20C3CF247D18",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.iphone-se-1",
               "modelName" : "iPhone SE (2nd generation)",
               "name" : "iPhone SE (2nd generation)"
             },
             {
               "modelCode" : "iPhone13,2",
               "simulator" : false,
               "modelName" : "iPhone 12",
               "error" : {
                 "code" : -13,
                 "failureReason" : "",
                 "underlyingErrors" : [
                   {
                     "code" : 4,
                     "failureReason" : "",
                     "description" : "Roman’s iPhone is locked.",
                     "recoverySuggestion" : "To use Roman’s iPhone with Xcode, unlock it.",
                     "domain" : "DVTDeviceIneligibilityErrorDomain"
                   }
                 ],
                 "description" : "Roman’s iPhone is not connected",
                 "recoverySuggestion" : "Xcode will continue when Roman’s iPhone is connected and unlocked.",
                 "domain" : "com.apple.platform.iphoneos"
               },
               "operatingSystemVersion" : "16.0 (20A362)",
               "identifier" : "00008101-000251120181001E",
               "platform" : "com.apple.platform.iphoneos",
               "architecture" : "arm64e",
               "interface" : "usb",
               "available" : false,
               "name" : "Roman’s iPhone",
               "modelUTI" : "com.apple.iphone-12-1"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad6,4",
               "identifier" : "3C5C27BC-4361-4AB2-9406-F80FD7C1426E",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.ipad-pro-9point7-a1674-b9b7ba",
               "modelName" : "iPad Pro (9.7-inch)",
               "name" : "iPad Pro (9.7-inch)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone12,1",
               "identifier" : "BF61D4D9-62EA-40BD-A81A-57D33FFBB5BF",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.iphone-11-1",
               "modelName" : "iPhone 11",
               "name" : "iPhone 11"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad13,18",
               "identifier" : "20A3F5D2-EFCF-4978-8A79-035621008BD8",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.ipad-10-1",
               "modelName" : "iPad (10th generation)",
               "name" : "iPad (10th generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone14,7",
               "identifier" : "51BEC6C1-CBB1-4DF0-B55A-2DBA9C2DEBFD",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.iphone-14-1",
               "modelName" : "iPhone 14",
               "name" : "iPhone 14"
             },
             {
               "simulator" : false,
               "operatingSystemVersion" : "13.0.1 (22A400)",
               "interface" : "usb",
               "available" : true,
               "platform" : "com.apple.platform.macosx",
               "modelCode" : "MacBookPro17,1",
               "identifier" : "00008103-001644840AE0801E",
               "architecture" : "arm64e",
               "modelUTI" : "com.apple.macbookpro-13-retina-touchid-late-2020",
               "modelName" : "MacBook Pro",
               "name" : "My Mac"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone15,3",
               "identifier" : "A122D9DA-8C82-48DA-BB7F-1BAE9F16E4B5",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.iphone-14-pro-max-1",
               "modelName" : "iPhone 14 Pro Max",
               "name" : "iPhone 14 Pro Max"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone8,4",
               "identifier" : "3A252E63-D6E4-4DBD-A680-7447F663362C",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.iphone-se-a1662-aeb1b8",
               "modelName" : "iPhone SE (1st generation)",
               "name" : "iPhone SE (1st generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad13,17",
               "identifier" : "2E6E90A8-99C5-4A2B-8D37-0C1D606F4128",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.ipad-air5-1",
               "modelName" : "iPad Air (5th generation)",
               "name" : "iPad Air (5th generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad7,12",
               "identifier" : "0CFC09D2-4ED4-4C32-86C7-F64C7CF1CDD4",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.ipad-7-wwan-1",
               "modelName" : "iPad (7th generation)",
               "name" : "iPad (7th generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPad8,9",
               "identifier" : "4EBED6A6-BBEC-47E2-BA29-595AB746ABF2",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.ipad-pro-11-2nd-1",
               "modelName" : "iPad Pro (11-inch) (2nd generation)",
               "name" : "iPad Pro (11-inch) (2nd generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "16.1 (20B72)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone14,6",
               "identifier" : "363B25BA-0D70-4DB4-B189-BB068E90E7F4",
               "architecture" : "arm64",
               "modelUTI" : "com.apple.iphone-se3-1",
               "modelName" : "iPhone SE (3rd generation)",
               "name" : "iPhone SE (3rd generation)"
             },
             {
               "simulator" : true,
               "operatingSystemVersion" : "13.7 (17H22)",
               "available" : true,
               "platform" : "com.apple.platform.iphonesimulator",
               "modelCode" : "iPhone12,3",
               "identifier" : "469E4F47-12CE-45E5-B6D3-C73BA791183D",
               "architecture" : "x86_64",
               "modelUTI" : "com.apple.iphone-11-pro-1",
               "modelName" : "iPhone 11 Pro",
               "name" : "iPhone 11 Pro"
             }
           ]
[   +6 ms] Error: iPhone (Roman) is not connected. Xcode will continue when iPhone (Roman) is connected and unlocked. (code -13)
[        ] Error: Roman’s iPhone is not connected. Xcode will continue when Roman’s iPhone is connected and unlocked. (code -13)
[   +2 ms] /Users/roman.cinis@caronsale.de/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell getprop
[  +49 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +39 ms] Skipping pub get: version match.
[  +86 ms] Launching lib/main.dart on Chrome in debug mode...
[  +40 ms] Initializing file store
[   +5 ms] Skipping target: gen_localizations
[   +1 ms] Skipping target: gen_dart_plugin_registrant
[        ] Skipping target: _composite
[        ] complete
[        ] Updating assets
[  +36 ms] Waiting for connection from debug service on Chrome...
[ +144 ms] <- reset
[   +2 ms] /Users/roman.cinis@caronsale.de/fvm/versions/beta/bin/cache/dart-sdk/bin/dart --disable-dart-dev /Users/roman.cinis@caronsale.de/fvm/versions/beta/bin/cache/dart-sdk/bin/snapshots/frontend_server.dart.snapshot --sdk-root /Users/roman.cinis@caronsale.de/fvm/versions/beta/bin/cache/flutter_web_sdk/ --incremental --target=dartdevc --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill /var/folders/md/vvs8jk413j383rgyrhqsskw80000gq/T/flutter_tools.Pk2dzC/flutter_tool.My0vA5/app.dill --packages /Users/roman.cinis@caronsale.de/Documents/GitHub/mixins_tests_bug/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-root /var/folders/md/vvs8jk413j383rgyrhqsskw80000gq/T/flutter_tools.Pk2dzC/flutter_tools.ruuBOs --filesystem-scheme org-dartlang-app --initialize-from-dill build/b1b715402d823b7fd5c2b68d2edcb2ce.cache.dill.track.dill --platform file:///Users/roman.cinis@caronsale.de/fvm/versions/beta/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk_sound.dill --verbosity=error --sound-null-safety
[   +5 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[ +239 ms] Waiting for connection from debug service on Chrome... (completed in 389ms)
[        ] Synced 0.0MB.
[        ] <- accept
[        ] Caching compiled dill
[  +26 ms] Using Google Chrome 108.0.5359.48 beta

[  +38 ms] executing: sysctl hw.optional.arm64
[   +3 ms] Exit code 0 from: sysctl hw.optional.arm64
[        ] hw.optional.arm64: 1
[   +6 ms] Found ARM Chrome installation at /Applications/Google Chrome Beta.app/Contents/MacOS/Google Chrome Beta, forcing native launch.
[ +339 ms] [CHROME]: 
[        ] [CHROME]: DevTools listening on ws://127.0.0.1:61828/devtools/browser/3cf1e36b-08f2-4605-855d-44bac272c0de
[ +475 ms] DwdsInjector: Received request for entrypoint at http://localhost:61824/main_module.bootstrap.js
[   +1 ms] MetadataProvider: Loading debug metadata...
[   +1 ms] MetadataProvider: Loaded debug metadata for module: web_entrypoint.dart
[        ] MetadataProvider: Loaded debug metadata for module: web_plugin_registrant.dart
[        ] MetadataProvider: Loaded debug metadata for module: packages/mixins_tests_bug/main.dart
[        ] MetadataProvider: Loaded debug metadata (sound null safety)
[        ] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:61824/main_module.bootstrap.js
[ +810 ms] ChromeProxyService: Initializing expression compiler for main_module.bootstrap.js with sound null safety: true
[  +22 ms] DevHandler: Debug service listening on ws://127.0.0.1:61846/Le0sannX4Lw=/ws

[   +9 ms] DevHandler: VmService proxy responded with an error:
           {jsonrpc: 2.0, id: 9, error: {code: -32601, message: Method not found, data: {jsonrpc: 2.0, method: _setStreamIncludePrivateMembers, id: 9, params: {streamId: Stdout, includePrivateMembers: false}}}}
[   +2 ms] This app is linked to the debug service: ws://127.0.0.1:61846/Le0sannX4Lw=/ws
[        ] DevHandler: VmService proxy responded with an error:
           {jsonrpc: 2.0, id: 10, error: {code: -32601, message: Method not found, data: {jsonrpc: 2.0, method: _setStreamIncludePrivateMembers, id: 10, params: {streamId: Stderr, includePrivateMembers: false}}}}
[        ] DevHandler: VmService proxy responded with an error:
           {jsonrpc: 2.0, id: 11, error: {code: -32601, message: Method not found, data: {jsonrpc: 2.0, method: _setStreamIncludePrivateMembers, id: 11, params: {streamId: Isolate, includePrivateMembers: false}}}}
[   +1 ms] DevHandler: VmService proxy responded with an error:
           {jsonrpc: 2.0, id: 12, error: {code: -32601, message: Method not found, data: {jsonrpc: 2.0, method: _setStreamIncludePrivateMembers, id: 12, params: {streamId: Extension, includePrivateMembers: false}}}}
[   +1 ms] Debug service listening on ws://127.0.0.1:61846/Le0sannX4Lw=/ws
[        ] 💪 Running with sound null safety 💪
[        ] 🔥  To hot restart changes while running, press "r" or "R".
[        ] For a more detailed help message, press "h". To quit, press "q".
[        ] An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:61846/Le0sannX4Lw=
```

```
Analyzing mixins_tests_bug...                                           
No issues found! (ran in 1.3s)
```

```
[✓] Flutter (Channel beta, 3.6.0-0.1.pre, on macOS 13.0.1 22A400 darwin-arm64, locale en-CZ)
    • Flutter version 3.6.0-0.1.pre on channel beta at /Users/roman.cinis@caronsale.de/fvm/versions/beta
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 75927305ff (3 days ago), 2022-11-17 05:52:50 +0530
    • Engine revision b754eabc98
    • Dart version 2.19.0 (build 2.19.0-374.1.beta)
    • DevTools version 2.19.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/roman.cinis@caronsale.de/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment Zulu17.36+17-CA (build 17.0.4.1+1-LTS)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Google Chrome Beta.app/Contents/MacOS/Google Chrome Beta

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.73.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.53.20221101

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 13 (API 33) (emulator)
    • macOS (desktop)             • macos         • darwin-arm64   • macOS 13.0.1 22A400 darwin-arm64
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 108.0.5359.48 beta

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by exaby73** [comment]
## Triage report

I can reproduce this issue. With `Stable (3.3.8)`, I get `90.9%` coverage while with `Beta (3.6.0-0.1.pre)` and `Master (3.6.0-6.0.pre.29)`, I get `45.5%` coverage.

<details>
<summary><code>lcov.info</code> Stable</summary>

```txt
SF:lib/proper_coverage.dart
DA:4,1
DA:5,1
DA:9,3
DA:10,1
DA:12,1
LF:5
LH:5
end_of_record
SF:lib/main.dart
DA:4,1
DA:5,1
DA:9,3
DA:10,1
DA:12,1
DA:16,0
LF:6
LH:5
end_of_record
```

</details>

<details>
<summary><code>lcov.info</code> Beta</summary>

```txt
SF:lib/proper_coverage.dart
DA:4,1
DA:5,1
DA:9,3
DA:10,1
DA:12,1
LF:5
LH:5
end_of_record
SF:lib/main.dart
DA:4,0
DA:5,0
DA:9,0
DA:10,0
DA:12,0
DA:16,0
LF:6
LH:0
end_of_record
```

</details>

**Comment by christopherfujino** [comment]
I'm marking this p2 for further triage by the dart team, not sure of the actual impact

**Comment by jason-simmons** [comment]
Reproduced this and traced it to https://dart.googlesource.com/sdk/+/86233b55bcc87832f32f944485d5cb841c564fee

@liamappelbe

**Comment by liamappelbe** [comment]
That CL is just a small optimisation, so I'll revert it for now

**Comment by liamappelbe** [comment]
Revert landed: https://dart-review.googlesource.com/c/sdk/+/273100

**Comment by exaby73** [comment]
@liamappelbe Has this also landed in the master branch? If so, this issue can be closed as fixed.

**Comment by a-siva** [comment]
I believe the above fix from Liam did not make it to the Dart 2.19 beta3 branch cut and so will have to be cherry picked.
